### ISSUE TITLE
`Type` isnt lowercase like with `list` and `dict

### ISSUE THREAD

**Issue by Zomatree** [microsoft/pyright#2467]
**Describe the bug**
similar to `list` and `dict`, python allows you to do `type[Foo]` the same as `typing.Type[Foo]`, in the pyright errors, pyright always shows `Type` not `type`, whereas pyright shows `list` and `List` depending on which is used

**To Reproduce**
```py
foo: type[int] = 1
```
```
Expression of type "Literal[1]" cannot be assigned to declared type "Type[int]"
```

**Expected behavior**
```
Expression of type "Literal[1]" cannot be assigned to declared type "type[int]"
```

**Comment by Zomatree** [comment]
> Type is an alias of type, and they can be used interchangeably with each other, so I don't consider this a bug.

`Type` and `List` are defined the exact same in the typing lib, `Type` and `type` cant be used for the same thing, `Type[int]()` is invalid in the same way `List[int]()` is invalid

**Comment by erictraut** [comment]
When used in a type annotation — and in error messages related to types, `Type` and `type` are interchangeable. 

From the perspective of a type checker, `Type` is a very special case, quite different from `List` and other simple aliases in many ways. If you look in `typing.pyi`, you can see that `List`, `Dict`, etc. are defined as an `_Alias`, whereas `Type` is a `_SpecialForm`.

What problem do you see with using `Type` in error messages? Do you find it to be confusing or ambiguous?
